org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Synchronizer.d

```diff
@@ -208,6 +213,9 @@ public void syncExec (Runnable runnable) {
             SWT.error (SWT.ERROR_FAILED_EXEC, lock.throwable);
         }
     }
+    // Release handle of lock#cond.
+    // If not released here, the handle will not be released until the next GC works.
+    destroy(lock);
```

Comment:
What happens when the lock is added here: https://github.com/d-widget-toolkit/dwt/blob/a4633cd433160c6688aecd2e3c4c51d3d406e5ae/org.eclipse.swt.win32.win32.x86/src/org/eclipse/swt/widgets/Synchronizer.d#L193 but then destroyed here.

Reply:
This `lock` is processed in the following order in `syncExec()`:

 1. Added to the message queue by `addLast()` (Synchronizer.d:Line 193).
 2. It is pop of the queue and executed by `runAsyncMessages()` called from the GUI thread (Line 141-). The thread of calling `syncExec()` waits for completion of processing (Line 200-).
 3. `lock` can be destroyed after the completion of the wait (Line 218).

`asyncExec()` has no wait. Must be destroyed in `runAsyncMessages()`.